homeassistant/components/mqtt/device_trigger.py

```diff
@@ -326,22 +392,41 @@ async def async_attach_trigger(
     trigger_info: TriggerInfo,
 ) -> CALLBACK_TYPE:
     """Attach a trigger."""
+    trigger_id: str | None = None
     mqtt_data = get_mqtt_data(hass)
     device_id = config[CONF_DEVICE_ID]
-    discovery_id = config[CONF_DISCOVERY_ID]
 
-    if discovery_id not in mqtt_data.device_triggers:
-        mqtt_data.device_triggers[discovery_id] = Trigger(
+    # The use of CONF_DISCOVERY_ID was deprecated in HA Core 2024.2.
```

Comment:
When a trigger is set up from an automation first, we'll respect the discovery_id.